config/v1/types_ingress.go

```diff
@@ -151,8 +151,22 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// network load balancer. Present only if type is NLB.
+	//
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
```

Comment:
Add a featuregate to this field.  It's done by 
1. add a featuregate like https://github.com/openshift/api/blob/master/config/v1/feature_gates.go#L182-L188
2. add a `// +openshift:enable:FeatureGate=Example` comment to this type
3. run `hack/update-payload-featuregates.sh`
4. run `make update`

This will add the schema to `TechPreviewNoUpgrade` (files now auto-generated).  Once your automated tests are passing or QE signs off, a PR enabling by default can flip the gate to true [ref](https://github.com/openshift/enhancements/blob/master/dev-guide/featuresets.md#id-like-to-declare-a-feature-accessible-by-default--what-is-the-process)

Reply:
OK. I need to try this. Thanks!

Reply:
`make verify` gives the following error:

```
+ '[' '!' -f /Users/misalunk/go/src/github.com/openshift/api/operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml ']'
+ echo 'No test suite file found for CRD /Users/misalunk/go/src/github.com/openshift/api/operator/v1/zz_generated.featuregated-crd-manifests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml, expected /Users/misalunk/go/src/github.com/openshift/api/operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml'
No test suite file found for CRD /Users/misalunk/go/src/github.com/openshift/api/operator/v1/zz_generated.featuregated-crd-manifests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml, expected /Users/misalunk/go/src/github.com/openshift/api/operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml
+ exit 1
make: *** [verify-scripts] Error 1
misalunk@misalunk-mac api %  
```

After creating that file `touch /Users/misalunk/go/src/github.com/openshift/api/operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml` `make verify` suceeds.

However, after running this command `./hack/update-payload-featuregates.sh` the `/Users/misalunk/go/src/github.com/openshift/api/operator/v1/tests/ingresscontrollers.operator.openshift.io/SetEIPForNLBIngressController.yaml` is deleted.

Reply:
done

Reply:
Better to move the feature gating up to this level, rather than the fields within. And you'll need to feature gate the new CEL validation as well

Reply:
added at both places